src/modules/kubernetes.rs

```diff
@@ -107,6 +107,15 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         return None;
     };
 
+    // We check that detect_env_vars is not empty here first since calling any on an amptry
+    // iternator returns false, since we want to preserve backwards compatibiliTy we default to
+    // being enabled.
+    let have_env_vars = !config.detect_env_vars.is_empty()
+        && config
+            .detect_env_vars
+            .iter()
+            .any(|e| context.get_env(e).is_some());
```

Comment:
This has a helper in `Context` now:
```suggestion
    let have_env_vars = context.detect_env_vars(&config.detect_env_vars);
```